src/lib/tuning/index/index_evaluator.cpp

```diff
@@ -0,0 +1,75 @@
+#include "index_evaluator.hpp"
+
+#include "optimizer/column_statistics.hpp"
+#include "optimizer/table_statistics.hpp"
+#include "resolve_type.hpp"
+#include "storage/index/base_index.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
+#include "utils/logging.hpp"
+
+namespace opossum {
+
+IndexEvaluator::IndexEvaluator() {}
+
+void IndexEvaluator::_setup() { _saved_work.clear(); }
+
+void IndexEvaluator::_process_access_record(const BaseIndexEvaluator::AccessRecord& record) {
+  auto table_statistics = StorageManager::get().get_table(record.column_ref.table_name)->table_statistics();
+  // ToDo(anyone) adapt for multi column indices...
+  auto predicate_statistics =
+      table_statistics->predicate_statistics(record.column_ref.column_ids[0], record.condition, record.compare_value);
+  auto total_rows = table_statistics->row_count();
+  auto match_rows = predicate_statistics->row_count();
+  auto unscanned_rows = total_rows - match_rows;
+  float saved_work = unscanned_rows * record.query_frequency;
```

Comment:
Why is this a float? Because you had other types of saved work in mind instead of rows?

Reply:
Exactly, as *desirability* can mean different things for different evaluators. Also in the case of the `IndexEvaluator`, `TableStatistics::row_count()` returns `float`.